TimescaleDB 2.6 release notes (#799)
timescale/docs: timescaledb/overview/release-notes/index.md

For our next releases, we plan to add:

- Performance improvements for:

>   - Inserts and common queries on Multi-node 

Reviewer: Is it for 2.7? We already have notable improvements in 2.6, maybe makes sense to mention them as well.

Author: We do have a list of them for 2.7 planned, yes.

Reviewer: So why are the performance improvements of 2.6 over 2.5 not mentioned in the release notes?

Reviewer: Added to the changelog https://github.com/timescale/timescaledb/pull/4084/commits/9f83a03a6ff1ecd5b781fc845640a2fedafba5c4

>   - Continuous Aggregates by removing re-aggregation 

Reviewer 2: ```suggestion
  - Continuous aggregates by removing re-aggregation 
```

> - Custom origin's support in continuous aggregates with `time_bucket_ng`

Reviewer 2: Is the indentation off here?

Author: The other 2 above are level 2.
This one is level 1.

For our next releases, we plan to add:

> TBD

Author: ```suggestion
- Performance improvements for:
  - Inserts and common queries on Multi-node 
  - Continuous Aggregates by removing re-aggregation 
- Custom origin's support in continuous aggregates with `time_bucket_ng`
```

on [community Slack](https://slack.timescale.com/) for installation details and
follow these [setup instructions][distributed-hypertables-setup].

### What's new in TimescaleDB 2.6:

- Continuous aggregates with compression
- `time_bucket_ng` support for N months and Timezones on continuous aggregates

You can read more about this release on our [blog post](https://tsdb.co/timescaledb-2-5).
This release also contains bug fixes since the 2.5.0 release.

 <highlight type="note"> This release is low priority for upgrade. We recommend that you upgrade when you can. </highlight> 

<!-- <highlight type="important"> This release is medium priority for upgrade. We recommend that you upgrade at the next available opportunity. </highlight> -->

<!-- <highlight type="warning"> This release is high priority for upgrade. We strongly recommend that you upgrade as soon as possible. </highlight> -->

> The experimental features in the 2.6 release are:
> 
> TBD

Author: ```suggestion
The new experimental features in the 2.6 release are:

- `time_bucket_ng` support for N months on continuous aggregates
`time_bucket_ng` support for Timezones on continuous aggregates
```

on [community Slack](https://slack.timescale.com/) for installation details and
follow these [setup instructions][distributed-hypertables-setup].

### What's new in TimescaleDB 2.6:

- Continuous aggregates with compression
- `time_bucket_ng` support for N months and Timezones on continuous aggregates

You can read more about this release on our [blog post](https://tsdb.co/timescaledb-2-5).
This release also contains bug fixes since the 2.5.0 release.

 <highlight type="note"> This release is low priority for upgrade. We recommend that you upgrade when you can. </highlight> 

<!-- <highlight type="important"> This release is medium priority for upgrade. We recommend that you upgrade at the next available opportunity. </highlight> -->

<!-- <highlight type="warning"> This release is high priority for upgrade. We strongly recommend that you upgrade as soon as possible. </highlight> -->

> The new experimental features in the 2.6 release are:

Author: ```suggestion
The experimental features in the 2.6 release are:
```

on [community Slack](https://slack.timescale.com/) for installation details and
follow these [setup instructions][distributed-hypertables-setup].

### What's new in TimescaleDB 2.6:

- Continuous aggregates with compression

> - `time_bucket_ng` support for N months and timezones on continuous aggregates

Author: ```suggestion
- `time_bucket_ng` support for N months, years and timezones on continuous aggregates
```

on [community Slack](https://slack.timescale.com/) for installation details and
follow these [setup instructions][distributed-hypertables-setup].

### What's new in TimescaleDB 2.6:

- Continuous aggregates with compression

> - `time_bucket_ng` support for N months and Timezones on continuous aggregates

Reviewer 3: We also support years according to https://github.com/timescale/timescaledb/pull/3753

Author: We support N months, years, n months and years... I believe it confuses the messaging.

> - `time_bucket_ng` support for N months and Timezones on continuous aggregates

Reviewer 2: ```suggestion
- `time_bucket_ng` support for N months and timezones on continuous aggregates
```

> 

Reviewer 2: ```suggestion
```

You can read more about this release on our [blog post](https://tsdb.co/timescaledb-2-5).
This release also contains bug fixes since the 2.5.0 release.

>  <highlight type="note"> This release is low priority for upgrade. We recommend that you upgrade when you can. </highlight> 

Author: ```suggestion
<!-- <highlight type="note"> This release is low priority for upgrade. We recommend that you upgrade when you can. </highlight> -->
```

<highlight type="important"> This release is medium priority for upgrade. We recommend that you upgrade at the next available opportunity. </highlight>

<!-- <highlight type="warning"> This release is high priority for upgrade. We strongly recommend that you upgrade as soon as possible. </highlight> -->

The experimental features in the 2.6 release are:

> 

Reviewer 2: ```suggestion
```

* The `time_bucket_ng` function, a newer version of `time_bucket`. This function
supports years, months, days, hours, minutes, seconds, and timezones.

> * `time_bucket_ng` support for N months and Timezones on continuous aggregates

Reviewer 3: same comment as above, we also support years

> <!-- <highlight type="important"> This release is medium priority for upgrade. We recommend that you upgrade at the next available opportunity. </highlight> -->

Author: ```suggestion
<highlight type="important"> This release is medium priority for upgrade. We recommend that you upgrade at the next available opportunity. </highlight>
```

## Release notes

> TBD
> 

Author: ```suggestion
## 2.6.0 (2022-02-16)
This release adds major new features since the 2.5.2 release, among
them:

* Compression in continuous aggregates
* Experimental support for timezones in continuous aggregates
* Experimental support for monthly buckets in continuous aggregates

It also includes several bug fixes. Telemetry reports are switched to a new
format, and now include more detailed statistics on compression, distributed
hypertables and indexes.  We deem this release to be moderate priority for
upgrading.

**Features**
* #3768 Allow ALTER TABLE ADD COLUMN with DEFAULT on compressed hypertable
* #3769 Allow ALTER TABLE DROP COLUMN on compressed hypertable
* #3943 Optimize first/last
* #3945 Add support for ALTER SCHEMA on multi-node
* #3949 Add support for DROP SCHEMA on multi-node

**Bugfixes**
* #3808 Properly handle `max_retries` option
* #3863 Fix remote transaction heal logic
* #3869 Fix ALTER SET/DROP NULL contstraint on distributed hypertable
* #3944 Fix segfault in add_compression_policy
* #3961 Fix crash in EXPLAIN VERBOSE on distributed hypertable
* #4015 Eliminate float rounding instabilities in interpolate
* #4019 Update ts_extension_oid in transitioning state
* #4073 Fix buffer overflow in partition scheme

**Thanks**
* @fvannee for reporting a first/last memory leak
* @mmouterde for reporting an issue with floats and interpolate

```

* Compression in continuous aggregates
* Experimental support for timezones in continuous aggregates
* Experimental support for monthly buckets in continuous aggregates

It also includes several bug fixes. Telemetry reports are switched to a new
format, and now include more detailed statistics on compression, distributed
hypertables and indexes.  We deem this release to be moderate priority for
upgrading.

**Features**
* #3768 Allow ALTER TABLE ADD COLUMN with DEFAULT on compressed hypertable
* #3769 Allow ALTER TABLE DROP COLUMN on compressed hypertable
* #3943 Optimize first/last
* #3945 Add support for ALTER SCHEMA on multi-node
* #3949 Add support for DROP SCHEMA on multi-node

**Bug fixes**
* #3808 Properly handle `max_retries` option
* #3863 Fix remote transaction heal logic
* #3869 Fix ALTER SET/DROP NULL contstraint on distributed hypertable
* #3944 Fix segfault in add_compression_policy
* #3961 Fix crash in EXPLAIN VERBOSE on distributed hypertable
* #4015 Eliminate float rounding instabilities in interpolate
* #4019 Update ts_extension_oid in transitioning state
* #4073 Fix buffer overflow in partition scheme

> 

Author: ```suggestion

**Improvements**

Query planning performance is improved for hypertables with a large number of chunks.

```

## Release notes

## 2.6.0 (2022-02-16)
This release adds major new features since the 2.5.2 release, including:
* Compression in continuous aggregates
* Experimental support for timezones in continuous aggregates
* Experimental support for monthly buckets in continuous aggregates

It also includes several bug fixes. Telemetry reports are switched to a new
format, and now include more detailed statistics on compression, distributed
hypertables and indexes.  We deem this release to be moderate priority for
upgrading.

> **Features**
> * #3768 Allow ALTER TABLE ADD COLUMN with DEFAULT on compressed hypertable
> * #3769 Allow ALTER TABLE DROP COLUMN on compressed hypertable
> * #3943 Optimize first/last
> * #3945 Add support for ALTER SCHEMA on multi-node
> * #3949 Add support for DROP SCHEMA on multi-node
> 

Author: ```suggestion
**Features**
* #3768 Allow ALTER TABLE ADD COLUMN with DEFAULT on compressed hypertable
* #3769 Allow ALTER TABLE DROP COLUMN on compressed hypertable
* #3873 Enable compression on continuous aggregates 
* #3943 Optimize first/last
* #3945 Add support for ALTER SCHEMA on multi-node
* #3949 Add support for DROP SCHEMA on multi-node
* #3977 Timezones support in CAGGs

```

## Release notes

## 2.6.0 (2022-02-16)

> This release adds major new features since the 2.5.2 release, among
> them:

Reviewer 2: ```suggestion
This release adds major new features since the 2.5.2 release, including:
```

> 

Reviewer 2: ```suggestion
```

* Compression in continuous aggregates
* Experimental support for timezones in continuous aggregates

> * Experimental support for monthly buckets in continuous aggregates

Reviewer 3: same comments as above, we also support years

Reviewer 4: Do you need to mention "years" here as well?

It also includes several bug fixes. Telemetry reports are switched to a new
format, and now include more detailed statistics on compression, distributed
hypertables and indexes.  We deem this release to be moderate priority for
upgrading.

**Features**
* #3768 Allow ALTER TABLE ADD COLUMN with DEFAULT on compressed hypertable
* #3769 Allow ALTER TABLE DROP COLUMN on compressed hypertable
* #3943 Optimize first/last
* #3945 Add support for ALTER SCHEMA on multi-node
* #3949 Add support for DROP SCHEMA on multi-node

> **Bugfixes**

Reviewer 2: ```suggestion
**Bug fixes**
```
